homeassistant/components/bosch_shc/alarm_control_panel.py

```diff
@@ -0,0 +1,95 @@
+"""Platform for alarm control panel integration."""
+from __future__ import annotations
+
+from boschshcpy import SHCIntrusionSystem, SHCSession
+
+from homeassistant.components.alarm_control_panel import (
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DATA_SESSION, DOMAIN
+from .entity import SHCDomainEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the alarm control panel platform."""
+    session: SHCSession = hass.data[DOMAIN][config_entry.entry_id][DATA_SESSION]
+
+    alarm_control_panel = IntrusionSystemAlarmControlPanel(
+        domain=session.intrusion_system,
+        parent_id=session.information.unique_id,
+        entry_id=config_entry.entry_id,
+    )
+
+    async_add_entities([alarm_control_panel])
+
+
+class IntrusionSystemAlarmControlPanel(SHCDomainEntity, AlarmControlPanelEntity):
```

Comment:
This should implement `has_entity_name` but it seems nowhere to be found.
I think we should have a preliminary PR fixing that first and then we can finalize this one.